ui/src/app/views/concept-survey-add-modal.tsx

```diff
@@ -0,0 +1,232 @@
+    this.getExistingSurveyConceptSets();
+  }
+
+  async getExistingSurveyConceptSets() {
+    try {
+      const {workspace: {namespace, id}} = this.props;
+      const conceptSets = await conceptSetsApi().getSurveyConceptSetsInWorkspace(namespace, id,
+        this.props.surveyName);
+      const conceptSetsInSurveys = conceptSets.items
+          .filter((conceptset) => conceptset.domain === Domain.OBSERVATION);
+
+      this.setState({
+        conceptSets: conceptSetsInSurveys,
+        addingToExistingSet: conceptSetsInSurveys.length > 0,
+        loading: false,
+      });
+      if (conceptSetsInSurveys) {
+        this.setState({selectedSet: conceptSetsInSurveys[0]});
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  async saveConcepts() {
+    const {workspace: {namespace, id}, onSave} = this.props;
+    const {selectedSet, addingToExistingSet, newSetDescription, name} = this.state;
+    this.setState({saving: true});
+    const conceptIds = this.props.selectedSurvey.map((surveys) => surveys.conceptId);
+    let survey = Surveys.THEBASICS;
+    switch (this.props.surveyName) {
+      case 'Lifestyle': survey = Surveys.LIFESTYLE; break;
+      case 'Overall Health': survey = Surveys.OVERALLHEALTH; break;
+      case 'The Basics': survey = Surveys.THEBASICS; break;
+      default: {console.error('Survey name not found'); return; }
+    }
+    if (addingToExistingSet) {
+      const updateConceptSetReq: UpdateConceptSetRequest = {
+        etag: selectedSet.etag,
+        addedIds: conceptIds
+      };
+      try {
+        const conceptSet = await conceptSetsApi().updateConceptSetConcepts(
+          namespace, id, selectedSet.id, updateConceptSetReq);
+        this.setState({saving: false});
+        onSave(conceptSet);
+      } catch (error) {
+        console.error(error);
+      }
+    } else {
+      const conceptSet: ConceptSet = {
+        name: name,
+        description: newSetDescription,
+        domain: Domain.OBSERVATION,
+        survey: survey
+      };
+      const request: CreateConceptSetRequest = {
+        conceptSet: conceptSet,
+        addedIds: conceptIds
+      };
+      try {
+        const createdConceptSet =
+           await conceptSetsApi().createConceptSet(namespace, id, request);
+        this.setState({saving: false});
+        onSave(createdConceptSet);
+      } catch (error) {
+        console.error(error);
+        this.setState({errorSaving: true});
+      }
+    }
+  }
+
+
+  render() {
+    const {onClose, selectedSurvey, surveyName} = this.props;
+    const {conceptSets, loading, name, nameTouched, addingToExistingSet} = this.state;
+    const errors = validate({name}, {
+      name: {
+        presence: {allowEmpty: false},
+        exclusion: {
+          within: conceptSets.map((concept: ConceptSet) => concept.name),
+          message: 'already exists'
+        }
+      }
+    });
+    return <Modal><ModalTitle data-test-id='add-concept-title'>
```

Comment:
this modal is almost exactly the same as what's in `concept-add-modal.tsx`.  Can we pull that out to a common component that's used in both places, rather than having duplicate code?

Reply:
Will create a new story for this